components/dashboard/src/usage/download/DownloadUsage.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC, useCallback, useState } from "react";
+import { Button } from "../../components/Button";
+import { downloadUsageCSV } from "./download-usage-csv";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+import { Dayjs } from "dayjs";
+import { useToast } from "../../components/toasts/Toasts";
+import { useCurrentOrg } from "../../data/organizations/orgs-query";
+
+type Props = {
+    attributionId: AttributionId;
+    startDate: Dayjs;
+    endDate: Dayjs;
+};
+export const DownloadUsage: FC<Props> = ({ attributionId, startDate, endDate }) => {
+    const { data: org } = useCurrentOrg();
+    const [isSaving, setIsSaving] = useState(false);
+    const { toast } = useToast();
+
+    const handleDownload = useCallback(async () => {
+        if (!org) {
+            return;
+        }
+
+        setIsSaving(true);
+
+        try {
+            const downloaded = await downloadUsageCSV({
+                orgName: org?.slug ?? org?.id,
+                attributionId: AttributionId.render(attributionId),
+                from: startDate.startOf("day").valueOf(),
+                to: endDate.endOf("day").valueOf(),
+            });
+
+            if (!downloaded) {
+                toast("There are no usage records for that date range");
+            }
+        } catch (e) {
+            console.error(e);
+            toast(`There was a problem downloading your usage data: ${e.message}`);
+        }
+
+        setIsSaving(false);
+    }, [attributionId, endDate, org, startDate, toast]);
+
+    return (
+        <Button type="secondary" loading={isSaving} onClick={handleDownload}>
+            Download as CSV
```

Comment:
**suggestion**: Adding here the optimized SVG code for the export/download icon.

```svg
<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" fill="none">
  <path fill="#A8A29E" d="M10.0208 12.6779c-.1282 0-.25605-.0258-.38355-.0775-.12751-.0516-.25179-.1385-.37283-.2607L6.17308 9.24842c-.21259-.21261-.31675-.43751-.31248-.67469.00427-.23716.11537-.46472.33332-.68266.21794-.21795.43936-.32212.66425-.3125.22489.00961.44631.12338.66427.34133l1.56089 1.58175V3.77244c0-.24679.09455-.46474.28365-.65383.1891-.1891.40704-.28365.65382-.28365.2468 0 .4647.09455.6538.28365.1891.18909.2837.40704.2837.65383v5.72921L12.54 7.9199c.2126-.2126.4327-.32504.6603-.33733.2275-.01228.4503.09055.6682.3085.218.21794.327.44416.327.67866 0 .2345-.109.46073-.327.67869l-3.0913 3.09128c-.121.1222-.2453.2091-.3728.2607-.1275.0517-.2554.0775-.3836.0775Zm-4.76438 4.2179c-.59768 0-1.10234-.2058-1.51396-.6174-.41164-.4117-.61746-.9163-.61746-1.514V13.5c0-.2468.09456-.4648.28367-.6539.18909-.189.40704-.2836.65383-.2836s.46474.0946.65383.2836c.1891.1891.28365.4071.28365.6539v1.2644c0 .0641.02671.1229.08012.1763.05343.0534.11221.0801.17632.0801h9.48708c.0642 0 .1229-.0267.1764-.0801.0534-.0534.0801-.1122.0801-.1763V13.5c0-.2468.0945-.4648.2836-.6539.1891-.189.4071-.2836.6539-.2836.2468 0 .4647.0946.6538.2836.1891.1891.2837.4071.2837.6539v1.2644c0 .5977-.2059 1.1023-.6175 1.514-.4116.4116-.9163.6174-1.514.6174H5.25642Z"/>
</svg>
```
Pasting below the export for previewing:

> ![export](https://github.com/gitpod-io/gitpod/assets/120486/61d8bb5f-b5b0-44e4-a494-2b41ed1689e1)